src/Database/QueryBuilder.php

```diff
@@ -65,14 +65,18 @@ public function cacheTags($cacheTags)
     }
 
     /**
-      * {@inheritDoc}
-      */
-     public function get($columns = ['*'])
-     {
+     * {@inheritDoc}
+     */
+    public function get($columns = ['*'])
+    {
         if (!is_null($this->cacheMinutes)) {
             return $this->getCached($columns);
         }
 
+        if (MemoryCache::instance()->enabled()) {
```

Comment:
Please move the MemoryCache check before the cacheMinutes check so that queries hit the memory cache before hitting the default cache. Once that's done we can merge this. Thanks for your work!